src/plugins/telemetry/server/routes/telemetry_opt_in.ts

```diff
@@ -109,7 +110,13 @@ export function registerTelemetryOptInRoutes({
         });
       }
 
-      await updateTelemetrySavedObject(context.core.savedObjects.client, attributes);
+      try {
+        await updateTelemetrySavedObject(context.core.savedObjects.client, attributes);
+      } catch (e) {
+        if (SavedObjectsErrorHelpers.isForbiddenError(e)) {
+          return res.forbidden();
+        }
```

Comment:
note: the legacy world relied on the `return res.forbidden()` response above when retrieving the telemetry saved object. Now that we require users to have `read` access to the telemetry object, we need to add a `forbidden` check here at update-time in order to get the expected response code for under-privileged users.